src/store/ProductsContext.js

```diff
@@ -0,0 +1,213 @@
+import {createContext,useEffect,useReducer} from 'react';
+import axios from 'axios';
+export const ProductsContext=createContext();
+
+export const ProductsContextProvider=({children})=>{
+    useEffect(()=>{
+        (async()=>{
+            const data=await axios.get("/api/products");
+            dispatch ({
+                type:"LOADPRODUCTLIST",
```

Comment:
For dispatch type use "LOAD_PRODUCT_LIST"  instead of "LOADPRODUCTLIST". It is more readable.